### PR TITLE
User roles backend

### PR DESCRIPTION
**Pull request by realgcoding**
* implemented UI for invite user , invited user table and user details update
* integrated invite user form and invite user table with backend

### CODE REVIEW

File: src/components/Dasboard/Cards/cardHooks/GetCardType.ts [hunk 1]
```diff
@@ -152,7 +152,10 @@ export function GetCardTypeAndValues(props: CardProps) {
 			pieCardConfig.loading = loading;
 			fetchedData = data;
 		}
-
+		console.log("fetched", fetchedData);
+		if (fetchedData === 0) {
```
Comment (ShadabFaiz) [inline src/components/Dasboard/Cards/cardHooks/GetCardType.ts:156]: `fectData === 0`. What does it mean?

File: src/components/Forms/UserRole/UserRole.tsx [hunk 2]
```diff
@@ -29,12 +35,29 @@ function UserRoleForm(props: UserRoleProps) {
...
+	useQuery(GET_ROLES_BY_ORG, {
 		variables: { filter: { organization: dashboardData?.organization?.id } },
 		onCompleted(data) {
-			if (data?.roles) {
-				userRoleForm[1].optionsArray = data.roles;
+			if (data?.organizationRoles) {
+				let roleArr: any = [];
+				data.organizationRoles.forEach((role: any) => {
+					/*excluding Admin role here from roles*/
+					if (role.type !== `admin-org-${dashboardData?.organization?.id}`) {
```
Comment (ShadabFaiz) [inline src/components/Forms/UserRole/UserRole.tsx:56]: why are we match role type with a string directly?

File: src/components/Forms/User/User.tsx [hunk 3]
```diff
@@ -80,7 +80,17 @@ function UserForm(props: UserProps) {
...
 			if (verifyAndUpdateUserForm) {
-				console.log("QUERRY");
+				updateUser({
+					variables: {
+						id: value.id,
+						input: {
+							name: value.name,
+							password: value.password,
+							profile_photo:
+								value.uploadPhoto === "removed" ? null : uploadResponse?.[0]?.id,
```
Comment (ShadabFaiz) [inline src/components/Forms/User/User.tsx:90]: why are we matching with a string directly?

**Review by ShadabFaiz — CHANGES_REQUESTED** [review]

File: src/components/Dasboard/Cards/cardHooks/GetCardType.ts (hunk 1 above)
Comment (realgcoding) [reply src/components/Dasboard/Cards/cardHooks/GetCardType.ts:156]: we get 0 value from backend if the list is empty .. it should be an empty array

Comment (realgcoding) [reply src/components/Dasboard/Cards/cardHooks/GetCardType.ts:156]: in GetBudgetCategory

File: src/components/Forms/UserRole/UserRole.tsx (hunk 2 above)
Comment (realgcoding) [reply src/components/Forms/UserRole/UserRole.tsx:56]: this is to remove admin role from list .. every role contains this type i.e admin-org-[organizationID] . i discussed this with amresh sir

File: src/components/Forms/User/User.tsx (hunk 3 above)
Comment (realgcoding) [reply src/components/Forms/User/User.tsx:90]: will change it to enum

**Pull request closed without merging**